src/AsyncStorage.native.js

```diff
@@ -322,6 +330,7 @@ const AsyncStorage = {
     callback?: ?(errors: ?$ReadOnlyArray<?Error>) => void,
   ): Promise<null> {
     return new Promise((resolve, reject) => {
+      checkValidArgs(arguments);
```

Comment:
We should check the arguments before creating the promise. Can we move this up a line?